fiftyone/core/collections.py

```diff
@@ -7726,6 +7726,20 @@ def count(self, field_or_expr=None, expr=None, safe=False):
         Returns:
             the count
         """
+
+        # Optimization: use estimated document count when possible
+        if (
+            isinstance(self, fod.Dataset)
```

Comment:
should fix it. the way groups are currently implemented, we cannot use estimated count.
```suggestion
            isinstance(self, fod.Dataset)
            and self.media_type != fom.GROUP
```

Reply:
I generalized it slightly more in https://github.com/voxel51/fiftyone/pull/5398/commits/d11868cb1e37fa2f6a06264a3895cc728879f131 so that estimated counts can also be used for grouped datasets when applicable, namely:

```py
grouped_dataset.select_group_slices().count()
```